### PR TITLE
[exs] Updated the app.js to export the file for use in index.js, which can be used when the app is deployed to a hosting environment.

### PR DESCRIPTION
**Pull request by zhouyaoji**

### CODE REVIEW

File: examples/developer-guide/adding_view_engines/app.js [hunk 1]
```diff
@@ -30,3 +30,4 @@ app.listen(app.get('port'), function () {
     debug('Server listening on port ' + app.get('port') + ' ' +
                'in ' + app.get('env') + ' mode');
 });
+module.exports = app;
```
Comment (imalberto) [inline examples/developer-guide/adding_view_engines/app.js:32]: New line please.

File: examples/developer-guide/binding_events/app.js [hunk 2]
```diff
@@ -30,3 +30,4 @@ app.listen(app.get('port'), function () {
     debug('Server listening on port ' + app.get('port') + ' ' +
                'in ' + app.get('env') + ' mode');
 });
+module.exports = app;
```
Comment (imalberto) [inline examples/developer-guide/binding_events/app.js:34]: New line.

File: examples/developer-guide/configure_routing/app.js [hunk 3]
```diff
@@ -30,3 +30,4 @@ app.listen(app.get('port'), function () {
     debug('Server listening on port ' + app.get('port') + ' ' +
                'in ' + app.get('env') + ' mode');
 });
+module.exports = app;
```
Comment (imalberto) [inline examples/developer-guide/configure_routing/app.js:32]: New line.

File: examples/developer-guide/dashboard/01_mojito_cli_basics/app.js [hunk 4]
```diff
@@ -26,3 +26,4 @@ app.listen(app.get('port'), function () {
     debug('Server listening on port ' + app.get('port') + ' ' +
                'in ' + app.get('env') + ' mode');
 });
+module.exports = app;
```
Comment (imalberto) [inline examples/developer-guide/dashboard/01_mojito_cli_basics/app.js:29]: New line.

File: examples/developer-guide/dashboard/01_mojito_cli_basics/index.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+var http = require('http'),
+    app = require('./app');
+
+/**
+ * @token given by manhattan and used to emit that the app is ready
+ */
+module.exports = function(config, token) {
+    // send the application to Manhattan along with the token
+    process.emit("application-ready", token, http.createServer(app));
+};
```
Comment (imalberto) [inline examples/developer-guide/dashboard/01_mojito_cli_basics/index.js:17]: New line.

File: examples/developer-guide/dashboard/02_mojits/app.js [hunk 6]
```diff
@@ -30,3 +30,4 @@ app.listen(app.get('port'), function () {
     debug('Server listening on port ' + app.get('port') + ' ' +
                'in ' + app.get('env') + ' mode');
 });
+module.exports = app;
```
Comment (imalberto) [inline examples/developer-guide/dashboard/02_mojits/app.js:32]: New line. Ok, last one. Please review the other `app.js`.

**Comment by imalberto** [comment]
@zhouyaoji 
Functional test `test_quickstartupguide_readpage_error.js` both failing on node8 and node10. Probably a typo on the example app during the renames.

Also, you might want to check PR#1342 to make sure that none of the `application.json` has a middleware configuration. See the PR doc for example, which needs to be added to the wiki.

File: examples/developer-guide/dashboard/02_mojits/app.js (hunk 6 above)
Comment (caridy) [inline examples/developer-guide/dashboard/02_mojits/app.js:32]: lol

Comment (imalberto) [inline examples/developer-guide/dashboard/02_mojits/app.js:32]: Learned from the best ! :)

**Comment by zhouyaoji** [comment]
@imalberto Sure, I'll add new lines to `app.js`, check the app causing that functional test to fail, and look for any middleware configuration in `application.json` files. Also, I realized that the `package.json` files for the examples are not correct either. I'll update the PR as soon as I can. Thanks!

**Pull request closed without merging**